tests/integration/linux_bridge_test.py

```diff
@@ -537,6 +539,37 @@ def test_disable_vlan_filtering_on_bridge(
             assertlib.assert_state_match(desired_state)
 
 
+def test_bridge_with_unmanaged_ports():
+    bridge_config = _create_bridge_subtree_config([])
+    with linux_bridge(TEST_BRIDGE0, bridge_config):
+        with dummy0_as_slave(TEST_BRIDGE0):
+            current_state = show_only((TEST_BRIDGE0,))
+            bridge_state = current_state[Interface.KEY][0]
+            port_subtree = bridge_state[LinuxBridge.CONFIG_SUBTREE].get(
+                LinuxBridge.PORT_SUBTREE, []
+            )
+
+            assert port_subtree[0].get(LinuxBridge.Port.NAME) == "dummy0"
+
+
+@contextmanager
+def dummy0_as_slave(master):
+    try:
+        exec_cmd(("ip", "link", "add", "dummy0", "type", "dummy"), check=True)
+        exec_cmd(("ip", "link", "set", "dummy0", "up"), check=True)
+        exec_cmd(
+            ("nmcli", "dev", "set", "dummy0", "managed", "no"), check=True
+        )
+        exec_cmd(("ip", "link", "set", "dummy0", "master", master), check=True)
+    except Exception:
+        exec_cmd(("ip", "link", "delete", "dummy0"))
+
```

Comment:
You need to raise here otherwise it will continue.
But I do not think you need this `Exception` block at all. If it fails, just get to the `finally` and that is it. One by try/finally block fits here, not two.

Reply:
If some of the commands fail then there is going to be a leftover. I am raising after removing the interface.

Reply:
You do this in the finally, that block is called for both regular exit and exceptions alike.

Reply:
Done, thanks!